frontend/src/employee-frontend/components/vasu/components/FollowupQuestion.tsx

```diff
@@ -193,24 +149,60 @@ export default React.memo(function FollowupQuestion({
       </QuestionInfo>
 
       <div>
-        {value.length > 0 ? (
-          value.map((entry, ix) => (
-            <FollowupEntryElement
+        {onChange ? (
+          entries.map(({ entry, date, textValue }, ix) => (
+            <FollowupEntryEditor
               key={ix}
               entry={entry}
-              onEdited={onEdited}
-              permittedFollowupActions={permittedFollowupActions}
+              date={date}
+              textValue={textValue}
+              data-qa={`follow-up-${ix}`}
+              onChange={(date, text) =>
+                onChange?.(
+                  entries
+                    .map((entry, i) =>
+                      i === ix
+                        ? entry.entry
+                          ? {
+                              ...entry.entry,
+                              date,
+                              text,
+                              edited:
+                                !entry.entry.createdDate ||
+                                HelsinkiDateTime.now().timestamp -
+                                  entry.entry.createdDate.timestamp >
+                                  15 * 60 * 1000
+                                  ? {
+                                      editorId: user?.id,
+                                      editorName: user?.name,
+                                      editedAt: LocalDate.todayInHelsinkiTz()
+                                    }
+                                  : entry.entry.edited
+                            }
+                          : text.length > 0
+                          ? {
+                              date,
+                              text,
+                              authorId: user?.id,
+                              authorName: user?.name,
+                              createdDate: HelsinkiDateTime.now()
+                            }
+                          : undefined
+                        : entry.entry
+                    )
+                    .filter((entry): entry is FollowupEntry => !!entry)
+                )
+              }
```

Comment:
I think it wouldn't hurt to use named functions or variables here to add more semantics for better readability